src/utils.test.ts

```diff
@@ -1,49 +1,47 @@
 import test from "ava"
-import { safeParseJSON } from "./utils"
-import { Buffer } from "buffer/"
+import { safeUserParse } from "./utils"
 
 let userNames = [
-  `【パンツ】Pantsu`,
-  `🎃 ASH  П (◣_◢) П  🎃`,
-  `七曜曜`,
-  `Dunesca ʕ•͡ᴥ•ʔ`,
+  `【パンツ】Pantsu#1234`,
+  `🎃 ASH  П (◣_◢) П  🎃#1234`,
+  `七曜曜#1234`,
+  `Dunesca ʕ•͡ᴥ•ʔ#1234`,
+  `Deadsiesthe•̪̀●́#1234`,
+  `Deadsiesthe"*�#1234`
 ]
 
-function toB64(str: string): string {
-  return Buffer.from(str, "ascii").toString("base64")
-}
-
-function fromB64(str: string): string {
-  return Buffer.from(str, "base64").toString()
-}
-
-function encode(obj: Record<string, unknown>): string {
-  return fromB64(toB64(JSON.stringify(obj)))
-}
-
 for (let user of userNames) {
-  test(`safeParseJSON can decode special user names: ${user}`, (t) => {
-    const TestValue = "This Should Be Returned"
-    let userString = encode({ TestValue, AccountName: user })
-    let decoded = safeParseJSON<{ AccountName: string; TestValue: string }>(
-      userString
-    )
+  test(`safeUserParse can parse special user names: ${user}`, (t) => {
+    const TestAccessToken = "This Should Be Returned"
+    let userString = `{\"AccessToken\": \"${TestAccessToken}\", \"AccountName\":\"${user}\"}`
+    let decoded = safeUserParse(userString)
     t.not(decoded, undefined)
-    t.is(decoded!.TestValue, TestValue)
+    t.is(decoded!.AccessToken, TestAccessToken)
+    t.not(decoded?.AccountName, undefined)
+    t.not(decoded?.AccountName, '')
+    if (decoded?.AccountName) {
+      t.is(/[\w-#]+/.test(decoded?.AccessToken), true)
+    }
```

Comment:
If you use the non-null assertion operator (https://www.typescriptlang.org/docs/handbook/2/everyday-types.html#non-null-assertion-operator-postfix-) you should be able to remove the `if`

```suggestion
    t.not(decoded!.AccountName, undefined)
    t.not(decoded!.AccountName, '')
    t.is(/[\w-#]+/.test(decoded!.AccessToken), true)
```

Not a fan of assertions but I'm happy with them in tests. If AccessToken was actually not on the object we would want the tests to fail anyway

Reply:
TIL - I think I've seen that once or twice before perhaps, but nice to see a use in the wild for it